pkg/scheduler/dependency_graph.go

```diff
@@ -381,11 +426,8 @@ func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.D
 	}
 
 	sort.Stable(&tmp)
-	var result []string
-	for _, replica := range tmp.Items {
-		result = append(result, replica.ReplicaName())
-	}
-	return result[:count], nil
+
+	return tmp.Items[:count], nil, nil
 }
 
 // BuildDependencyGraph loads dependencies data and creates the DependencyGraph
```

Comment:
This func grown into too big.
At last https://github.com/istalker2/k8s-AppController/blob/54b8e5aff01184904606d4a929576752c984ca84/pkg/scheduler/dependency_graph.go#L423 to https://github.com/istalker2/k8s-AppController/blob/54b8e5aff01184904606d4a929576752c984ca84/pkg/scheduler/dependency_graph.go#L465 can be extracted to separate named block.

Big plus for extracting `fillDependencyGraph` which imo still needs simplification.